### PR TITLE
Refactor/forms validation

### PR DESCRIPTION
**Pull request by shanetzulinho**

### CODE REVIEW

File: src/components/nutrient/NutrientForm.js [hunk 1]
```diff
@@ -65,102 +86,7 @@ const NutrientForm = ({ getMetabolizableEnergy, getDryMatterBasis, getCalciumRat
...
-            type="number"
-            InputProps={{ inputProps: { min: 0, step: 0.1 } }}
-            {...register('phosphorus', {
-              required: 'Phosphorus (磷) is required.',
-              valueAsNumber: true,
-            })}
-            error={Boolean(errors.phosphorus)}
-            helperText={errors.phosphorus?.message}
-          />
+          {renderInputs}
```
Comment (iamseye) [inline src/components/nutrient/NutrientForm.js:89]: 👍 
You can directly render here
```
{ Object.values(INPUTS).map((key) =>  (
      <Controller
        key={key}
        name={INPUTS_ATTRIBUTES[key].name}
        control={control}
        rules={validationNumberRegisterOptions}
        defaultValue=""
        render={({ field }) => (
          <InputField
            {...field}
            label={INPUTS_ATTRIBUTES[key].label}
            type="number"
            error={Boolean(errors[key])}
            helperText={errors[key]?.message}
          />
        )}
      />
    ))
   }
  ```

File: src/components/calories/CaloriesForm.js [hunk 2]
```diff
@@ -41,27 +42,38 @@ const CaloriesForm = ({ getLowerUpperBound }) => {
...
-            items={activityLevelOptions}
-            {...register('DREFactor', {
-              required: 'Activity Level (活動量) is required.',
-            })}
-            error={Boolean(errors.DREFactor)}
-            helperText={errors.DREFactor?.message}
+            control={control}
+            defaultValue=""
+            rules={{
+              required: 'This field is required (此為必填欄位)',
```
Comment (iamseye) [inline src/components/calories/CaloriesForm.js:65]: Hi, Maybe you can have rules for the valid required fields like `validationNumberRegisterOptions` and use them here. it seems not so consistent, line 48 uses attributes, but here are hardcode the rules.

**Review by iamseye — APPROVED** [review]
Looks good~ just have some small comments :)

File: src/components/nutrient/NutrientForm.js (hunk 1 above)
Comment (shanetzulinho) [reply src/components/nutrient/NutrientForm.js:89]: I've made the change! Thanks for your advice! 👍

File: src/components/calories/CaloriesForm.js (hunk 2 above)
Comment (shanetzulinho) [reply src/components/calories/CaloriesForm.js:65]: I've made the change! Thanks 👍

**Pull request merged**